src/ImmersedBoundaries/conditional_fluxes.jl

```diff
@@ -16,17 +15,18 @@ Return either
 
 This can be used either to condition intrinsic flux functions, or immersed boundary flux functions.
 """
-@inline conditional_flux(i, j, k, ibg, ℓx, ℓy, ℓz, qᴮ, qᴵ, nc=ipn) = ifelse(nc(ℓx, ℓy, ℓz, i, j, k, ibg), qᴮ, qᴵ)
+@inline conditional_flux(i, j, k, ibg, ℓx, ℓy, ℓz, qᴮ, qᴵ) =
+    ifelse(immersed_peripheral_node(ℓx, ℓy, ℓz, i, j, k, ibg), qᴮ, qᴵ)
```

Comment:
@tomchor this is the key change to fix type inference (and then there is a cascade of associated changes in all the functions that use this operator).

Early in this PR I generalized `conditional_flux` to take an final optional positional argument `nc` (stands for `node_conditional`) to which one might pass either `immersed_peripheral_node` or `peripheral_node`.

I believe what was happening is that the compiler then refused to _specialize_ on this final argument (not sure why --- sometimes the compiler refuses to specialize on _function arguments_ because every function is it's own type, meaning that the relative advantage of specializing on an object which can only ever have just one instance is low). So the type inference algorithm stopped at this function. This would then prevent the _entire_ tendency calculation from being inlined, leading to a dramatic loss in performance.